packages/starlight/__tests__/basics/config-errors.test.ts

```diff
@@ -253,3 +253,32 @@ test('errors with sidebar entry that includes `items` and `autogenerate`', () =>
 			**sidebar.0**: Unrecognized key(s) in object: 'autogenerate'"
 	`);
 });
+
+test('parses route middleware config successfully', () => {
+	const data = parseStarlightConfigWithFriendlyErrors({
+		title: '',
+		routeMiddleware: './src/routeData.ts',
+	});
+	expect(data.routeMiddleware).toEqual(['./src/routeData.ts']);
+});
+
+test("errors if one route middleware is conflicting path with Astro' middleware", () => {
+	expect(() =>
+		parseStarlightConfigWithFriendlyErrors({
+			title: 'Test',
+			routeMiddleware: ['src/middleware.ts', 'src/routeData.ts'],
```

Comment:
This should be updated for the new RegEx:

```suggestion
			routeMiddleware: ['./src/middleware.ts', './src/routeData.ts'],
```